### ISSUE TITLE
IDG for V2L models

### ISSUE THREAD

**Issue by wj210** [chasewalker26/Integrated-Decision-Gradients#1]
Hi, thanks for the great work!

May i ask if IDG is possible to be used with VisiontoText models such as BLIP-2?
Given an image an query question, i like to use IDG to find the important pixels in the image.

**Comment by chasewalker26** [comment]
Hello, thank you! Yes, IDG is fully model agnostic, so you can use it here. At the most basic level, all that is needed to compute IDG is model gradients and the model logits or confidence of prediction. Ideally, you would compute the gradients which relate the input text and image pixels to the output text, and then use IDG's interpolation method to improve the quality of these gradients as stated in the paper. I recommend starting with implementing IG for the BLIP model and then making the simple changes needed to implement IDG. Please see https://github.com/chasewalker26/Integrated-Decision-Gradients/blob/main/util/attribution_methods/saliencyMethods.py for a simple IG implementation that should be easy to follow. Hopefully this helps!

**Comment by wj210** [comment]
Hi, thank you for the prompt response!
I managed to make some edits for it to work with v2l models.
